src/main/java/com/google/devtools/build/lib/vfs/DigestHashFunction.java

```diff
@@ -82,8 +73,6 @@ private DigestHashFunction(
     checkArgument(!names.isEmpty());
     this.name = names.get(0);
     this.names = names;
-    this.messageDigestPrototype = getMessageDigestInstance();
-    this.messageDigestPrototypeSupportsClone = supportsClone(messageDigestPrototype);
   }
 
   public static DigestHashFunction register(
```

Comment:
We cannot change how the registration works here because we rely on this internally. You could, however, glue BLAKE3 into this with following steps:

1. Write a class that implements `java.security.MessageDigest` for `BLAKE3`. This is the class that interact with JNI and call into c code.
2. Register the `MessageDigest` from step 1 by implementing a `java.security.Provider` and add it with `java.security.Security.addProvider(...)`. So that when you call `MessageDigest.getInstance("BLAKE3")`, you can get back the instance from the provider.
3. Adapt `DigestMessage` for `BLAKE3` into `HashFunction` by calling [MessageDigestHashFunction](https://cs.opensource.google/guava/guava/+/master:guava/src/com/google/common/hash/MessageDigestHashFunction.java;l=47;drc=e62d6a0456420d295089a9c319b7593a3eae4a83).
4. Create a `DigestHashFunction` by calling ` DigestHashFunction.register(...)` using the `HashFunction` from step 3.

I know this is verbose and not fun but we have to write it in this way.